src/audio/aria/aria_generic.c

```diff
@@ -40,10 +40,9 @@ inline void aria_algo_calc_gain(struct aria_data *cd, size_t gain_idx,
 		src = audio_stream_wrap(source, src + n);
 		samples -= n;
 	}
-
 	/*zero check for maxis not needed since att is in range <0;3>*/
-	if (max_data > (0x7fffffff >> att))
-		gain = (0x7fffffffULL << 32) / max_data;
+	if (max_data > (0x007fffff >> att))
```

Comment:
there is definition for int24, you can reuse it:
src/include/sof/audio/format.h:33:#define INT24_MAXVALUE  8388607
src/include/sof/audio/format.h:34:#define INT24_MINVALUE -8388608